CIP-Registration/CIP-Registration.md

```diff
@@ -0,0 +1,101 @@
+---
+CIP: TODO
+Title: Catalyst Registration Transaction Metadata Format
+Authors: Sebastien Guillemot <sebastien@emurgo.io>, Rinor Hoxha <rinor.hoxha@iohk.io>, Mikhail Zabaluev <mikhail.zabaluev@iohk.io>
+Comments-URI: https://forum.cardano.org/t/cip-catalyst-registration-metadata-format/44038
+Status: Draft
+Type: Standards
+Created: 2020-01-05
+License: CC-BY-4.0
+---
+
+## Abstract
+
+Cardano uses a sidechain for its treasury system. One needs to "register" to participate on this sidechain by submitting a registration transaction on the mainnet chain. This CIP details the registration transaction format.
+
+## Motivation
+
+Cardano uses a sidechain for its treasury system ("Catalyst"). One of the desirable properties of this sidechain is that even if its safety is compromised, it doesn't cause loss of funds on the main Cardano chain. To achieve this, instead of using your wallet's recovery phrase on the sidechain, we need to use a brand new "voting key".
+
+However, since 1 ADA = 1 vote, a user needs to associate their mainnet ADA to their new voting key. This can be achieved through a registration transaction.
+
+We therefore need a registration transaction that serves two purposes:
+
+1. Registers a "voting key" to be included in the sidechain
+2. Associates mainnet ADA to this voting key
+
+## Specification
+
+### Voting key format
+
+A voting key is simply an ED25519 key. How this key is created is up to the wallet.
+
+### Associating stake with a voting key
+
+Recall: Cardano uses the UTXO model so to completely associate a wallet's balance with a voting key (i.e. including enterprise addresses), we would need to associate every payment key to a voting key individually. Although there are attempts at this (see [CIP8](../CIP-0008/CIP-0008.md)), the resulting data structure is a little excessive for on-chain metadata (which we want to keep small)
+
+Given the above, we choose to only associate staking keys with voting keys. Since most Cardano wallets only use base addresses for Shelley wallet types, in most cases this should perfectly match the user's wallet.
+
+### Registration metadata format
+
+A Catalyst registration transaction are a regular Cardano transaction with a specific transaction metadata associated with it
+
+Notably, there should be two entries inside the metadata map:
+
+Voting key registration
+```
+61284: {
+  // voting_key - CBOR byte array
+  1: "0xa6a3c0447aeb9cc54cf6422ba32b294e5e1c3ef6d782f2acff4a70694c4d1663",
+  // stake_pub - CBOR byte array
+  2: "0xad4b948699193634a39dd56f779a2951a24779ad52aa7916f6912b8ec4702cee"
```

Comment:
For catalyst fund3 we are adding a `3:``which is a payment address in raw bytes. If you have control of the keys that generated this content, you probably want to use them to generate the address. It's the 0th index of the wallet in Daedalus. This is so we can pay voter rewards to that instead of having to do 1000's of MIR's from treasury we can just have 1 per fund.``
```suggestion
  2: "0xad4b948699193634a39dd56f779a2951a24779ad52aa7916f6912b8ec4702cee"
  3: "0x00588e8e1d18cba576a4d35758069fe94e53f638b6faf7c07b8abd2bc5c5cdee47b60edc7772855324c85033c638364214cbfc6627889f81c4"
```